src/hooks/init/buy-phone-number.js

```diff
@@ -0,0 +1,182 @@
+    const NewCommandClass = class extends TwilioClientCommand {
+      async runCommand() {
+        const phoneNumber = await this.getPhoneNumber();
+
+        if (!await this.confirmPurchase(phoneNumber)) {
+          throw new TwilioCliError('Cancelled');
+        }
+
+        await this.purchasePhoneNumber(phoneNumber);
+      }
+
+      async getPhoneNumber() {
+        const listCommandRunner = new ApiCommandRunner(
+          this.twilioClient,
+          listCommand.actionDefinition,
+          listCommand.flags,
+          this.flags
+        );
+
+        const response = await listCommandRunner.run();
+
+        this.logger.debug(`Search returned ${response.length} phone number(s)`);
+
+        if (response.length === 0) {
+          throw new TwilioCliError('Search returned 0 phone numbers. Please expand your search criteria.');
+        }
+
+        const limitedData = this.getLimitedData(response, this.flags.properties);
+        const outputLines = OutputFormats.columns(response, limitedData).split('\n');
+        const outputHeader = outputLines[0];
+        const phoneNumberLines = outputLines.slice(1);
+
+        const choices = response.map((item, index) => ({
+          name: phoneNumberLines[index],
+          value: item.phoneNumber
+        }));
+
+        const phoneNumberSelection = await this.inquirer.prompt([{
+          type: 'list',
+          name: 'phoneNumber',
+          // Throw on a newline (and space) so the selected phone number shows on the next line.
+          message: outputHeader + '\n ',
+          choices: choices,
+          pageSize: 10 // Seems like a reasonable size.
+        }]);
+
+        this.logger.debug(`Selected phone number: ${phoneNumberSelection.phoneNumber}`);
+
+        return phoneNumberSelection.phoneNumber;
+      }
+
+      async confirmPurchase(phoneNumber) {
+        const confirm = await this.inquirer.prompt([{
+          type: 'confirm',
+          name: 'affirmative',
+          message: `Are you sure you want to purchase the phone number "${phoneNumber}"?`
+        }]);
+
+        return confirm.affirmative;
+      }
+
+      async purchasePhoneNumber(phoneNumber) {
+        this.flags = {
+          [PHONE_NUMBER_FLAG]: phoneNumber
+        };
+
+        const createCommandRunner = new ApiCommandRunner(
+          this.twilioClient,
+          createCommand.actionDefinition,
+          createCommand.flags,
+          this.flags
+        );
+
+        const response = await createCommandRunner.run();
+
+        this.logger.info('Phone number successfully purchased!');
+
+        // Print the create response using the command's default output properties.
+        this.output(response, createCommand.flags.properties.default);
+      }
+    };
+
+    NewCommandClass.id = [this.commandTopic.name, phoneNumberType].join(TOPIC_SEPARATOR);
+    NewCommandClass.description = `purchase a ${capitalize(camelCase(phoneNumberType))} phone number`;
```

Comment:
The results of this function is a bit off to me, though I do like the automation (e.g. the former is the current output and the latter is what seems more readable to me):

purchase a Local phone number -> purchase a local phone number
purchase a MachineToMachine phone number -> purchase a machine to machine phone number
purchase a Mobile phone number -> purchase a mobile phone number
purchase a National phone number -> purchase a national phone number
purchase a SharedCost phone number -> purchase a shared cost phone number
purchase a TollFree phone number -> purchase a toll-free phone number
purchase a Voip phone number -> purchase a VoIP phone number

Reply:
How about this?
```
purchase a local phone number
purchase a machine-to-machine phone number
purchase a mobile phone number
purchase a national phone number
purchase a shared-cost phone number
purchase a toll-free phone number
purchase a voip phone number
```

Best I can do without special logic.